lib/enumerize/base.rb

```diff
@@ -98,17 +101,17 @@ def _set_default_value_for_enumerized_attributes
           value_for_validation = _enumerized_values_for_validation[attr.name.to_s]
 
           if (!attr_value || attr_value.empty?) && (!value_for_validation || value_for_validation.empty?)
-            value = attr.default_value
-
-            if value.respond_to?(:call)
-              value = value.arity == 0 ? value.call : value.call(self)
-            end
-
+            value = _call_if_callable(attr.default_value)
             public_send("#{attr.name}=", value)
           end
         rescue ActiveModel::MissingAttributeError
         end
       end
     end
+
+    def _call_if_callable(value)
```

Comment:
The module this method defined in is included in ActiveRecord::Base so methods can conflict with other libraries. I can suggest the following options that would allow to avoid any possible naming conflicts:

- Change the method name to something like `_enumerize_call_if_callable`.
- Simply repeat these few bits of code where its needed.
- Extract a Enumerize::Utils module with a singleton method so it can be used as `Utils.call_if_callable`.

I believe we should avoid any possible naming conflicts as much as possible. However, I don't have a strong preference what to choose, so feel free to use any approach you like more.

Reply:
Extract Enumerize::Utils module